src/buffer/out/Row.cpp

```diff
@@ -538,28 +538,87 @@ catch (...)
 
 [[msvc::forceinline]] void ROW::WriteHelper::ReplaceText() noexcept
 {
+    // This function starts with a fast-pass for ASCII. ASCII is still predominant in technical areas.
+    //
+    // We can infer the "end" from the amount of columns we're given (colLimit - colBeg),
+    // because ASCII is always 1 column wide per character.
+    auto it = chars.begin();
+    const auto end = it + std::min<size_t>(chars.size(), colLimit - colBeg);
     size_t ch = chBeg;
 
-    for (const auto& s : til::utf16_iterator{ chars })
+    while (it != end)
+    {
+        if (*it >= 0x80) [[unlikely]]
+        {
+            _replaceTextUnicode(ch, it);

```

Comment:
just so i understand, this takes in the new `ch` and the existing `colEnd` from the Helper and produces the correct outcome at the end?

Reply:
If it's working correctly, yeah. 😄 `ch` and `it` are the only state of the ASCII loop, so the Unicode loop needs to know about it.